code/obj/machinery/medimulcher.dm

```diff
@@ -0,0 +1,157 @@
+///How much food, by bites_left, the medimulcher can hold in its 'stomach' at once
+#define MDM_MAX_STOMACH 120
+///How many digested nutrients (1 per bites_left) the medimulcher needs to produce an anticyst (healing "patch") for you
+#define MDM_CYST_COST 35
+///Amount of digested nutrients the medimulcher can hold at one time
+#define MDM_MAX_DIGESTION 160
+
+///A prototypist "artifact reverse-engineer" reward: hybridized Martian tech turning food into synthflesh "patches"
+/obj/machinery/medimulcher
+	name = "\improper APS-4 bio-integrator"
+	desc = "According to the label, eats food and turns it into medicine. According to your eyes, possibly a crime against nature."
+	icon = 'icons/obj/manufacturer.dmi'
+	icon_state = "medimulcher"
+	anchored = ANCHORED
+	density = 1
+	flags = NOSPLASH
+	power_usage = 600
+	deconstruct_flags = DECON_SCREWDRIVER | DECON_WRENCH | DECON_CROWBAR
+
+	///Current amount of held food.
+	var/stomach = 0
+	///Cumulative progress - can pop out 1 anticyst per 100 progress.
+	var/digestion_progress = 0
+
+	//Overlay sprites
+	var/image/status_light = null
+	var/image/fill_bar = null
+
+	///"Stop poking me" tracker
+	var/last_poke_time = null
+
+	///Rolling counter to facilitate responsive overlay updates
+	var/last_ratio = 0
+
+	New()
+		..()
+		src.status_light = image('icons/obj/manufacturer.dmi', "")
+		src.fill_bar = image('icons/obj/manufacturer.dmi', "")
+		SPAWN(0)
+			src.overlay_refresh()
+
+	proc/overlay_refresh()
+		if (!(status & NOPOWER) && !(status & BROKEN))
+			if(digestion_progress >= MDM_CYST_COST) //ready to dispense? green light
+				src.status_light.icon_state = "mdm-stat-ready"
+			else if(stomach > 0) //not ready to dispense, but fed? yellow light
+				src.status_light.icon_state = "mdm-stat-working"
+			else //not ready to dispense, no food, but powered? red light
+				src.status_light.icon_state = "mdm-stat-empty"
+
+			var/ratio = min(1, src.stomach / MDM_MAX_STOMACH)
+			ratio = round(ratio, 0.2) * 100
+			src.fill_bar.icon_state = "mdm-fill-[ratio]"
+
+			src.UpdateOverlays(src.status_light, "statlight")
+			src.UpdateOverlays(src.fill_bar, "fillbar")
+		else
+			src.UpdateOverlays(null, "statlight")
+			src.UpdateOverlays(null, "fillbar")
+
+	attack_hand(mob/user)
+		src.add_fingerprint(user)
+
+		if (src.digestion_progress >= MDM_CYST_COST && !(status & (BROKEN|NOPOWER)))
+			src.digestion_progress -= MDM_CYST_COST
+			var/obj/item/I = new /obj/item/reagent_containers/patch/synthetic
+			playsound(src.loc, 'sound/impact_sounds/Glub_2.ogg', 80, 1)
+			boutput(user, SPAN_NOTICE("[src] dispenses a small glob.[prob(10) ? " Grody." : null]"))
+			user.put_in_hand_or_drop(I)
+		else
+			if(ishuman(user) && last_poke_time && (last_poke_time + 1 SECOND > world.time))
```

Comment:
```suggestion
			if(ishuman(user) && (last_poke_time + 1 SECOND > TIME))
```
also the time-based conditional seems... not useful here? esp considering you already have a cooldown below

Reply:
Its function is keeping you from randomly getting eaten if you just click it once without realizing it doesn't have anything to dispense, and don't repeatedly prod it trying to get it to do something.

Reply:
(Particularly important given you can lose a limb to it now.)